### PR TITLE
Use rawurlencode() to transform the Cookie into a string

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #23255
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by ro0NL — APPROVED** [review]

**Comment by javiereguiluz** [comment]
@nicolas-grekas I'va made that change too. However, when copying the `testToString()` test from HttpFoundation, I needed to fix all tests because the results are pretty different from the HttpFoundation tests. Are those differences expected?

Examples:

```php
$cookie = new Cookie('foo', 'bar', strtotime('Fri, 20-May-2011 15:25:52 GMT'), '/', '.myfoodomain.com', true);

// in HttpFoundation
(string) $cookie = 'foo=bar; expires=Fri, 20 May 2011 15:25:52 GMT; domain=.myfoodomain.com; path=/; secure; httponly'

// in BrowserKit
(string) $cookie = 'foo=bar; expires=Fri, 20-May-2011 15:25:52 GMT; path=/; domain=.myfoodomain.com; secure; httponly'
```

```php
$cookie = new Cookie('foo', 'bar', 0, '/', '');

// in HttpFoundation
(string) $cookie = 'foo=bar; path=/; httponly'

// in BrowserKit
(string) $cookie = 'foo=bar; expires=Thu, 01 Jan 1970 00:00:00 GMT; path=/; httponly'
```

File: src/Symfony/Component/BrowserKit/Cookie.php [hunk 1]
```diff
@@ -58,11 +58,11 @@ class Cookie
     public function __construct($name, $value, $expires = null, $path = null, $domain = '', $secure = false, $httponly = true, $encodedValue = false)
     {
         if ($encodedValue) {
-            $this->value = urldecode($value);
+            $this->value = rawurldecode($value);
```
Comment (ro0NL) [inline src/Symfony/Component/BrowserKit/Cookie.php]: if we update decoding, `Cookie::fromString` in HttpFoundation should probably be updated as well, note however `urldecode` works for `rawurlencode` perfectly fine, so im not sure we should change it (given `rawurldecode doesnt work for `urlencode`).

**Comment by stof** [comment]
@javiereguiluz your first example outputs the same cookie. Attributes are just given in a different order. But order is not meaningful.

**Comment by javiereguiluz** [comment]
@stof yes, the first example is the same, but with a different ordering. But I think it should be exactly the same. The attention a project pays to details like this says a lot about a project 😄

File: src/Symfony/Component/BrowserKit/Cookie.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/BrowserKit/Cookie.php]: works for me, we can keep urldecode here

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @javiereguiluz.

**Pull request closed without merging**